### PR TITLE
Vietnamese words support

### PR DESCRIPTION
**Pull request by duyet**
Currently worder doesn't support Vietnamese or other language with unicode. 
Ex from the simple test: 

``` js
worder("Tôi là Duyệt");
// [ 'T', 'i', 'l', 'Duy', 't' ]
```

And now is: 

``` js
worder("Tôi là Duyệt");
// [ 'Tôi', 'là', 'Duyệt' ]
```

### CODE REVIEW

**Comment by IonicaBizau** [comment]
Oh, awesome! I was wondering that `\W` takes care of that. :smile:

Isn't there a simpler way to handle this? For example, in Romanian we have these special characters: `ă`, `â`, `î`, `ș`, `ț` (and of course, uppercase).

Can we somehow catch 'em all in one rule or we have to specify each character?

File: lib/index.js [hunk 1]
```diff
@@ -8,5 +8,6 @@
  * @return {Array} An array of strings representing the text words.
  */
 module.exports = function worder(s) {
-    return s.split(/\W/g).filter(Boolean);
+    returnvar pattern = /[^a-z0-9A-Z_ÀÁÂÃÈÉÊÌÍÒÓÔÕÙÚĂĐĨŨƠàáâãèéêìíòóôõùúăđĩũơƯĂẠẢẤẦẨẪẬẮẰẲẴẶẸẺẼỀỀỂưăạảấầẩẫậắằẳẵặẹẻẽềếềểỄỆỈỊỌỎỐỒỔỖỘỚỜỞỠỢỤỦỨỪễệỉịọỏốồổỗộớờởỡợụủứừỬỮỰỲỴÝỶỸửữựỳỵỷỹ]+/;
```
Comment (IonicaBizau) [inline lib/index.js]: Typo: `returnvar`? Also, I guess, adding the `i` (case insensitive) flag would help.

**Comment by duyet** [comment]
The problem is that Javascript does not support Unicode regular expressions to token words.
However, a rather nice library called [XRegExp](http://xregexp.com).
How do you think about that :D ?

**Comment by IonicaBizau** [comment]
> The problem is that Javascript does not support Unicode regular expressions to token words.

Then, I'm thinking about splitting by spaces or making it a little bit smarter than a ~~simple~~ long complicated regex. Would you like to do that? Don't forget to add yourself as contributor in `package.json` as well. :grin: 

> How do you think about that :D ?

Looks big enough not to include it here. :smile:

File: lib/index.js (hunk 1 above)
Comment (duyet) [inline lib/index.js]: Sorry, It's mime mistake.

Comment (IonicaBizau) [inline lib/index.js]: :+1:

**Comment by IonicaBizau** [comment]
Merging this, and improvements like I said above are still welcome! :smile:

**Comment by duyet** [comment]
Splitting by spaces can be wrong sometimes. 

`Hello, world!` may become 

``` js
[ "Hello,", "world!" ]
```

Thus, we still need to filter non-characters. 
I thing making a smarter complicated regex is a great solution :smile:

**Comment by IonicaBizau** [comment]
> Thus, we still need to filter non-characters. 

Exactly! I guess taking out the grammar characters (question and exclamation marks, dots, etc) would be better.

> I thing making a smarter complicated regex is a great solution

That means we would have to support all the letters from all the languages, right? Maybe not that bad, tho... :smile:

**Comment by duyet** [comment]
Let's do it! :+1:  :smile:

**Comment by IonicaBizau** [comment]
Great! :smile: Any of the above suggestions can be implemented... Not sure which one is better. Choose the one you like more. :grin:

**Comment by duyet** [comment]
I will do some research. You are very friendly :smile:

**Comment by IonicaBizau** [comment]
Great :+1: Yes, I am! :smile: So happy when people contribute to my stuff. It's a special feeling. :sparkles:

**Comment by duyet** [comment]
:smile:

**Pull request merged**